book-content/chapters/07-mutability.md

```diff
@@ -0,0 +1,1013 @@
+In our chapter on unions and narrowing, we explored how TypeScript can infer types from the logical flow of our code. In this chapter, we'll see how mutability - whether a value can be changed or not - can affect type inference.
+
+## Mutability and Inference
+
+### Variable Declaration and Type Inference
+
+How you declare your variables in TypeScript affects whether or not they can be changed.
+
+#### How TypeScript Infers `let`
+
+When using the `let` keyword, the variable is _mutable_ and can be reassigned.
+
+Consider this `AlbumGenre` type: a union of literal values representing possible genres for an album:
+
+```ts twoslash
+type AlbumGenre = "rock" | "country" | "electronic";
+```
+
+Using `let`, we can declare a variable `albumGenre` and assign it the value `"rock"`. Then we can attempt to pass `albumGenre` to a function that expects an `AlbumGenre`:
+
+```ts twoslash
+// @errors: 2345
+type AlbumGenre = "rock" | "country" | "electronic";
+
+// ---cut---
+let albumGenre = "rock";
+
+const handleGenre = (genre: AlbumGenre) => {
+  // ...
+};
+
+handleGenre(albumGenre);
+```
+
+Because `let` was used when declaring the variable, TypeScript understands that the value can later be changed. In this case, it infers `albumGenre` as a `string` rather than the specific literal type `"rock"`. In our code, we could do this:
+
+```ts twoslash
+let albumGenre = "rock";
+
+// ---cut---
+albumGenre = "country";
+```
+
+Therefore, it will infer a wider type in order to accommodate the variable being reassigned.
+
+We can fix the error above by assigning a specific type to the `let`:
+
+```ts twoslash
+// @errors: 2345
+type AlbumGenre = "rock" | "country" | "electronic";
+
+// ---cut---
+let albumGenre: AlbumGenre = "rock";
+
+const handleGenre = (genre: AlbumGenre) => {
+  // ...
+};
+
+handleGenre(albumGenre); // no more error
+```
+
+Now, `albumGenre` _can_ be reassigned, but only to a value that is a member of the `AlbumGenre` union. So, it will no longer show an error when passed to `handleGenre`.
+
+But there's another interesting solution.
+
+#### How TypeScript Infers `const`
+
+When using `const`, the variable is _immutable_ and cannot be reassigned. When we change the variable declaration to use `const`, TypeScript will infer the type more narrowly:
```

Comment:
i wouldnt be surprised if somebody would take away from this that objects declared using `const` are immutable too but that's not true. I'd adjust the lingo used here to avoid mentioning immutability.